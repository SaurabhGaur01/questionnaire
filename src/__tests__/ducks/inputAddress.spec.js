import {
    setAddress,
    clearAddress,
    inputAddressReducer,
} from '../../ducks/userInput/inputAddress';

describe('inputAddress Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setAddress should match the snapshot' , () => {
            expect(setAddress(mockObject)).toMatchSnapshot();
        });
        it('clearAddress should match the snapshot' , () => {
            expect(clearAddress()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(inputAddressReducer()).toEqual("");
        });
        it('should return the passed state if called with no action' , () => {
            expect(inputAddressReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(inputAddressReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(inputAddressReducer('', setAddress(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearAddress' , () => {
            expect(inputAddressReducer('', clearAddress())).toEqual("");
        });
    });
});