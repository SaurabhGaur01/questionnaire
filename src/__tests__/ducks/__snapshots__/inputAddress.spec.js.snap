// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputAddress Duck Tests Action Generator clearAddress should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_ADDRESS",
}
`;

exports[`inputAddress Duck Tests Action Generator setAddress should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/SET_ADDRESS",
  "value": "mockValue",
}
`;
